shell.php

```diff
@@ -72,6 +81,18 @@ function featureShell($cmd, $cwd) {
     );
 }
 
+function featureCurrentUser() {
+    $username = isRunningWindows() ? getenv('USERNAME') : posix_getpwuid(posix_geteuid())['name'];
+    if ($username === false) {
```

Comment:
as `posix_getpwuid()` returns false if the user does not exist, `posix_getpwuid(...)['name']` will raise a warning and the value will be `null` so it will not take the default.

I updated the function like this before merging:

```php
function initShellConfig() {
    global $SHELL_CONFIG;

    if (isRunningWindows()) {
        $username = getenv('USERNAME');
        if ($hostname !== false) {
            $SHELL_CONFIG['username'] = $username;
        }
    } else {
        $pwuid = posix_getpwuid(posix_geteuid());
        if ($pwuid !== false) {
            $SHELL_CONFIG['username'] = $pwuid['name'];
        }
    }

    $hostname = gethostname();
    if ($hostname !== false) {
        $SHELL_CONFIG['hostname'] = $hostname;
    }
}
```

Reply:
If for whatever reason the returned uid is not really mapped to an existing user, this function can return false. (it should not happen in practice, but it is theoretically possible, especially when playing with confined processes or broken containers)